src/Symfony/Component/AssetMapper/ImportMap/ImportMapRenderer.php

```diff
@@ -125,11 +125,10 @@ public function render(string|array $entryPoint, array $attributes = []): string
             }
 
             $output .= <<<HTML
-                <script async$scriptAttributes>
+                <script$scriptAttributes>
                 if (!HTMLScriptElement.supports || !HTMLScriptElement.supports('importmap')) (function () {
                     const script = document.createElement('script');
                     script.src = '{$this->escapeAttributeValue($polyfillPath, \ENT_NOQUOTES)}';
-                    script.setAttribute('async', 'async');
```

Comment:
This is not what Mozilla doc states 🤔 

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#notes

> Scripts without [async](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#async), [defer](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#defer) or type="module" attributes, as well as inline scripts without the type="module" attribute, **are fetched and executed immediately** before the browser continues to parse the page.


And this seems to match html spec chart here

![HTMLSPEC](https://html.spec.whatwg.org/images/asyncdefer.svg)


Not found either in [the spec](https://html.spec.whatwg.org/multipage/scripting.html#dom-script-async) something that would suggest async is default (but i may have missed it)

Reply:
It’s from https://html.spec.whatwg.org/multipage/scripting.html#script-force-async:

> A [`script`](https://html.spec.whatwg.org/multipage/scripting.html#the-script-element) element has a **force async** boolean, initially true. It is set to false by the [HTML parser](https://html.spec.whatwg.org/multipage/parsing.html#html-parser) […] on [`script`](https://html.spec.whatwg.org/multipage/scripting.html#the-script-element) elements [it inserts], and when the element gets an [`async`](https://html.spec.whatwg.org/multipage/scripting.html#attr-script-async) content attribute added.

The polyfill script is not parser-inserted, but MDN only mentions these :confused: 

You can easily check a script-inserted script is async by default though: https://jsfiddle.net/nL509xph/

Reply:
> It is set to false by the [HTML parser](https://html.spec.whatwg.org/multipage/parsing.html#html-parser) and the [XML parser](https://html.spec.whatwg.org/multipage/xhtml.html#xml-parser) on [script](https://html.spec.whatwg.org/multipage/scripting.html#the-script-element) elements they insert

Reply:
> The polyfill script is not parser-inserted

Did you check [the fiddle](https://jsfiddle.net/nL509xph/)?

Reply:
I did check the fiddle :) 

I'm just not entirely sure this is enough to state how the browsers will behave. 

Your fiddle code "console log" attributes... this is not granted to be the behaviour off the parsers when they will start to load this (external) script.

And I think you will agree the MDN docs claims are something that creates doubts here.

I'm thinking except if we have a 100.00% insurance the script will _always_ be loaded after the main JS files, the async attribute in the generated script does not cost us much, is perfectly W3 valid, and could be a safe bet.

:)

Reply:
If you don’t trust the browsers nor the spec I fear you cannot be sure of anything.

But then why would we trust the W3 validator? And are we really 100% sure removing the `async` attribute from the inline script won’t have any unattended consequence?

Reply:
> If you don’t trust the browsers nor the spec I fear you cannot be sure of anything.

Not what i tried to say, nevermind :) If you're sure .. i don"t need to be convinced here 😅 

My point was just a small fear I had this would have unwanted consequences.. that's all :)